src/Bridge/Doctrine/Orm/SubresourceDataProvider.php

```diff
@@ -175,10 +175,22 @@ private function buildQuery(array $identifiers, array $context, QueryNameGenerat
                     $mappedBy = $classMetadata->getAssociationMapping($previousAssociationProperty)['mappedBy'];
                     $previousAlias = "$previousAlias.$mappedBy";
 
+                    $qb->select($alias)
+                        ->from($identifierResourceClass, $alias);
+                    break;
+                case ClassMetadataInfo::ONE_TO_ONE:
+                    $association = $classMetadata->getAssociationMapping($previousAssociationProperty);
+                    if (!isset($association['mappedBy'])) {
+                        goto def;
```

Comment:
Do we really want to use `goto`?

Reply:
the best thing to do here to avoid code duplication. see https://github.com/api-platform/core/pull/2228#discussion_r222557201

Reply:
@soyuka any preference on this? use goto or duplicate code in the if...

Reply:
I really don't know haha goto seemed good to me

Reply:
I'm also in favor of dropping the goto.